src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -223,9 +226,7 @@ var DOMRenderer = ReactFiberReconciler({
     rootContainerInstance : Container,
     internalInstanceHandle : Object,
   ) : void {
-    if (shouldAutoFocusHostComponent(type, newProps)) {
-      (domElement : any).focus();
-    }
+    ((domElement : any) : HTMLInputElement).focus();
```

Comment:
I guess this could be `HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement`.